lxd/network/network_utils_sriov.go

```diff
@@ -318,20 +318,52 @@ func SRIOVFindRepresentorPort(nicEntries []fs.DirEntry, pfSwitchID string, vfID
 			continue // Skip interfaces with no physical port name.
 		}
 
-		var pfID, nicVFID int
-		_, err = fmt.Sscanf(string(physPortName), "pf%dvf%d", &pfID, &nicVFID)
+		var nicPFID, nicVFID int
+		_, err = fmt.Sscanf(string(physPortName), "pf%dvf%d", &nicPFID, &nicVFID)
 		if err != nil {
 			continue // Skip non-VF interfaces.
 		}
 
-		if nicVFID == vfID {
+		if nicPFID == pfID && nicVFID == vfID {
 			return nic.Name() // We have a match.
 		}
 	}
 
 	return ""
 }
 
+func SRIOVGetSwitchAndPFID(parentDev string) (string, int, error) {
+	physPortName, err := os.ReadFile(filepath.Join(sysClassNet, parentDev, "phys_port_name"))
+	if err != nil {
+		return "", -1, err // Skip non-physical ports.
+	}
+
+	// Check the port is a physical port and not an existing representor port connected to the bridge
+	// but beloning to a physical device. This avoids trying to find a free VF repeatedly for the same
```

Comment:
`s/beloning/belonging/`

That typo was already there it seems ;)

Reply:
Ha, verbatim copying code around is not always safe ;) Thank you for pointing it out, will push a commit to correct the typo.

Reply:
That's the only thing I could review from that PR ;) The real deal will have to go through Tom.